sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -174,4 +176,89 @@ class CollationSuite extends QueryTest with SharedSparkSession {
           Row(expected))
     }
   }
+
+  test("create table with collation") {
+    val tableName = "parquet_dummy_tbl"
+    val collationName = "UCS_BASIC_LCASE"
+    val collationId = CollationFactory.collationNameToId(collationName)
+
+    withTable(tableName) {
+      sql(
+        s"""
+           |CREATE TABLE $tableName (c1 STRING COLLATE '$collationName')
+           |USING PARQUET
+           |""".stripMargin)
+
+      sql(s"INSERT INTO $tableName VALUES ('aaa')")
+      sql(s"INSERT INTO $tableName VALUES ('AAA')")
+
+      checkAnswer(sql(s"SELECT DISTINCT COLLATION(c1) FROM $tableName"), Seq(Row(collationName)))
+      assert(sql(s"select c1 FROM $tableName").schema.head.dataType == StringType(collationId))
+      val dd = sql(s"SELECT c1 FROM $tableName").toDF("c2")
+      dd.write.mode(SaveMode.Overwrite).parquet("/tmp/stefan_tbl")
```

Comment:
please use `withTempPath` to generate the temp path and write.

Reply:
whoops, I was supposed to delete that